lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -482,6 +483,11 @@ def configure(conf)
 
       @platform = detect_platform
 
+      # Treat an empty setting of the credentials file path environment variable
+      # as unset. This way the googleauth lib could fetch the credentials
+      # following the fallback path.
+      ENV.delete(CREDENTIALS_PATH_ENV_VAR) if ENV[CREDENTIALS_PATH_ENV_VAR] == ''
```

Comment:
[Optional] `ENV.delete(CREDENTIALS_PATH_ENV_VAR) unless ENV[CREDENTIALS_PATH_ENV_VAR].empty?`.

Reply:
Sync'ed offline. There was a typo. Igor in fact meant:
```
ENV.delete(CREDENTIALS_PATH_ENV_VAR) unless 
  ENV[CREDENTIALS_PATH_ENV_VAR].empty?
```

The problem with that is if `ENV[CREDENTIALS_PATH_ENV_VAR]` is nil, `.empty?` will error out.
```
$ irb
2.4.3 :001 > ENV['abc'].empty?
NoMethodError: undefined method `empty?' for nil:NilClass
        from (irb):1
        from /usr/local/google/home/lingshi/.rvm/rubies/ruby-2.4.3/bin/irb:11:in `<main>'
```

Reply:
Yeah...